packages/realm/src/app-services/App.ts

```diff
@@ -217,8 +217,13 @@ export class App<FunctionsFactoryType = DefaultFunctionsFactory, CustomDataType
     return currentUser ? User.get(currentUser) : null;
   }
 
-  public get allUsers(): User<FunctionsFactoryType, CustomDataType>[] {
-    return this.internal.allUsers.map((user) => User.get(user));
+  public get allUsers(): Readonly<Record<string, User<FunctionsFactoryType, CustomDataType>>> {
+    const result: Record<string, User<FunctionsFactoryType, CustomDataType>> = {};
+    for (const user of this.internal.allUsers) {
+      result[user.identity] = User.get(user);
+    }
+
+    return result;
   }
```

Comment:
Note: `realm-web` uses `Object.fromEntries()`. Feel free to comment on what is preferred here.

Reply:
Lol - I just saw this now 🤦 If I have to choose, I probably like `fromEntries` because it's shorter (fewer lines and doesn't need the type declared on `result`) and I generally like a pure function over iterating and mutating an object (the `result` in this case).

Reply:
Gotcha, despite the extra iteration from needing to create the array (`.map`) passed to `fromEntries`?